private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go

```diff
@@ -55,6 +56,24 @@ func checkEnumNoDelete(add addFunc, corpus *corpus, previousFile bufprotosource.
 	return nil
 }
 
+// CheckEnumSameType is a check function.
+var CheckEnumSameType = newEnumPairCheckFunc(checkEnumSameType)
+
+func checkEnumSameType(add addFunc, _ *corpus, previousEnum bufprotosource.Enum, enum bufprotosource.Enum) error {
+	previousDescriptor, err := previousEnum.AsDescriptor()
+	if err != nil {
+		return err
+	}
+	descriptor, err := enum.AsDescriptor()
+	if err != nil {
+		return err
+	}
+	if !previousDescriptor.IsClosed() && descriptor.IsClosed() {
+		add(enum, nil, withBackupLocation(enum.Features().EnumTypeLocation(), enum.Location()), `Enum %q changed from open to closed.`, enum.Name())
```

Comment:
Do we know that enum.Features() != nil? Can figure out myself later but thought would be quicker to ask

Reply:
Yes.